### PR TITLE
openal: bump openal/1.21.1

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **openal/1.21.1**

bump

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by danimtb** [comment]
Can't figure out what is happening with this PR. The error states "could not checkout" but the trace is confusing:
```
Command "git checkout -f 237c54e6439a057394cc4dac989cc90d9d403557" returned status code 1:
stdout: 
stderr: error: unable to create file recipes/openal/all/patches/1.21.0-c++14-does-not-have-std::aligned_alloc.patch: Invalid argument
Note: checking out '237c54e6439a057394cc4dac989cc90d9d403557'.
```
Maybe it is related to Windows workers not cleaning properly the workspace? 🤔

**Comment by madebr** [comment]
Perhaps the colon is causing a problem?
It is a reserved character: https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file

I'll rename the patch.

File: recipes/openal/all/patches/1.21.0-c++14-does-not-have-std-aligned_alloc.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
+gcc-11 triggers an error. std::aligned_alloc needs c++17.
+This patch ports openal 1.21.1 behavior back to 1.21.0.
+--- common/almalloc.cpp
++++ common/almalloc.cpp
+@@ -21,4 +21,4 @@
+-#if defined(HAVE_STD_ALIGNED_ALLOC)
```
Comment (SSE4) [inline recipes/openal/all/patches/1.21.0-c++14-does-not-have-std-aligned_alloc.patch:6]: it's c++17, but why HAVE_STD_ALIGNED_ALLOC is defined? shouldn't it detect the availability of the function?
I guess it should be fixed in another place

Comment (madebr) [reply recipes/openal/all/patches/1.21.0-c++14-does-not-have-std-aligned_alloc.patch:6]: openal is compiled as c++14.
I guess it was fine for gcc-10 but gcc-11 is causing problems.

Comment (madebr) [reply recipes/openal/all/patches/1.21.0-c++14-does-not-have-std-aligned_alloc.patch:6]: https://github.com/kcat/openal-soft/blob/62e6ce6d7322bc83ba3150316d9c921f5fadbb99/CMakeLists.txt#L156-L159

Comment (SSE4) [reply recipes/openal/all/patches/1.21.0-c++14-does-not-have-std-aligned_alloc.patch:6]: this seems to be an upstream fix?
https://github.com/kcat/openal-soft/commit/f576a353638236775ecb4f4e8ba93cb1716fbe2d

Comment (madebr) [reply recipes/openal/all/patches/1.21.0-c++14-does-not-have-std-aligned_alloc.patch:6]: Looks like it. I didn't look at the commit history and looked only at the diff around the `alogned_alloc`

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Pull request merged**